integration-tests/clipper_metric.py

```diff
@@ -28,33 +29,77 @@ def feature_sum(xs):
     return [str(sum(x)) for x in xs]
 
 
-logging.basicConfig(
-    format='%(asctime)s %(levelname)-8s [%(filename)s:%(lineno)d] %(message)s',
-    datefmt='%y-%m-%d:%H:%M:%S',
-    level=logging.INFO)
+def get_metrics_config():
+    config_path = os.path.join(
+        os.path.abspath("%s/../monitoring" % cur_dir), 'metrics_config.yaml')
+    with open(config_path, 'r') as f:
+        conf = yaml.load(f)
+    return conf
+
+
+def get_matched_query(metric_name):
+    query = gen_match_query(metric_name)
+    logger.info("Querying: {}".format(query))
+    res = requests.get(query).json()
+    logger.info(res)
+    return res
+
+
+def parse_res_and_assert_node(res, node_num):
+    assert res['status'] == 'success'
+    assert len(res['data']) == node_num
+
+
+def log_docker_ps():
+    logger.info(subprocess.check_output(['docker', 'ps']).decode('ascii'))
```

Comment:
Use the [Python Docker SDK](https://docker-py.readthedocs.io/en/stable/) not subprocess

Reply:
Fixed.